js/dcf-toggleButton.js

```diff
@@ -0,0 +1,297 @@
+        }, false);
+
+        // Show/hide element when the 'space' key is pressed
+        // Hide element when the `escape` is pressed
+        toggleButton.addEventListener('keydown', (keydownEvent) => {
+
+            this.toggleKeys.forEach((key) => {
+                if (DCFUtility.isKeyEvent(keydownEvent, DCFUtility.keyEvents(key))) {
+                    console.log("toggle", key)
+                    keydownEvent.preventDefault();
+                    this.toggleSwitched(toggleButton, toggleElement);
+                }
+            });
+            this.onKeys.forEach((key) => {
+                if (DCFUtility.isKeyEvent(keydownEvent, DCFUtility.keyEvents(key))) {
+                    console.log("on", key)
+                    keydownEvent.preventDefault();
+                    this.toggleSwitched(toggleButton, toggleElement, "open");
+                }
+            });
+            this.offKeys.forEach((key) => {
+                if (DCFUtility.isKeyEvent(keydownEvent, DCFUtility.keyEvents(key))) {
+                    console.log("off", key)
+                    keydownEvent.preventDefault();
+                    this.toggleSwitched(toggleButton, toggleElement, "close");
+                }
+            });
+        }, false);
+    }
+
+    // Handles the logic for the button
+    // This will only call the animations
+    toggleSwitched(toggleButton, toggleElement, state=""){
+
+        // Gets the labels for the button
+        const toggleButtonLabelOn = toggleButton.dataset.labelOn;
+        const toggleButtonLabelOff = toggleButton.dataset.labelOff;
+
+        // Toggled On
+        if (toggleButton.ariaExpanded === "false" && (state === "open" || state === "")) {
+            toggleButton.setAttribute('aria-expanded', 'true');
+            toggleButton.setAttribute('aria-label', toggleButtonLabelOff);
+            toggleButton.dispatchEvent(this.toggleButtonOn);
+
+            toggleElement.setAttribute('aria-hidden', 'false');
+            toggleElement.classList.remove('dcf-invisible', 'dcf-opacity-0', 'dcf-pointer-events-none');
+            toggleElement.classList.add('dcf-opacity-1', 'dcf-pointer-events-auto');
+            toggleElement.dispatchEvent(this.toggleElementOn);
+            toggleElement.focus();
+
+        // Toggle Off
+        } else if (toggleButton.ariaExpanded === "true" && (state === "close" || state === "")) {
+            toggleButton.setAttribute('aria-expanded', 'false');
+            toggleButton.setAttribute('aria-label', toggleButtonLabelOn);
+            toggleButton.dispatchEvent(this.toggleButtonOff);
+
+            toggleElement.setAttribute('aria-hidden', 'true');
+            toggleElement.classList.remove('dcf-opacity-1', 'dcf-pointer-events-auto');
+            toggleElement.classList.add('dcf-pointer-events-none');
+            // toggleElement.classList.add('dcf-opacity-0', 'dcf-invisible');
+            toggleElement.dispatchEvent(this.toggleElementOff);
+        }
+    }
+}
```

Comment:
No newline at end of file

Reply:
Added new line and fixed other simple lint issues